tests/interfaces.py

```diff
@@ -408,6 +415,82 @@ def test_GetSession_InvalidTokens(self):
         body = utils.check_response_basic_info(resp, 200)
         utils.check_val_equal(body["authenticated"], False)
 
+    @runner.MAGPIE_TEST_LOGIN
+    def test_LoginAnonymous_Forbidden(self):
+        """
+        Test different login variations and ensure that ``MAGPIE_ANONYMOUS_USER`` is always blocked.
+        """
+        utils.warn_version(self, "Login anonymous explicitly blocked.", "3.9.0", skip=True)
+        data = {
+            "user_name": get_constant("MAGPIE_ANONYMOUS_USER"),
+            "password": get_constant("MAGPIE_ANONYMOUS_PASSWORD")
+        }
+        headers = {"Accept": CONTENT_TYPE_JSON}
+
+        resp = utils.test_request(self, "GET", s.SigninAPI.path, params=data, headers=headers, expect_errors=True)
+        utils.check_response_basic_info(resp, 403, expected_method="GET")
+
+        form = "user_name={user_name}&password={password}".format(**data)
+        headers["Content-Type"] = CONTENT_TYPE_FORM
+        resp = utils.test_request(self, "POST", s.SigninAPI.path, data=form, expect_errors=True)
+        utils.check_response_basic_info(resp, 403, expected_method="POST")
+
+        headers["Content-Type"] = CONTENT_TYPE_JSON
+        resp = utils.test_request(self, "POST", s.SigninAPI.path, json=data, expect_errors=True)
+        utils.check_response_basic_info(resp, 403, expected_method="POST")
+
+    @runner.MAGPIE_TEST_LOGIN
+    def test_Login_GetRequestFormat(self):
+        """
+        Validate that GET request for login returns correct metadata, and not its corresponding POST request ones.
+
+        Tests multiple accept header formats to ensure they are also handled each time.
+        """
+        utils.warn_version(self, "Login with GET metadata validation.", "3.9.0", skip=True)
+
+        self.login_admin()
+        utils.TestSetup.delete_TestUser(self)
+        utils.TestSetup.delete_TestGroup(self)
+        utils.TestSetup.create_TestGroup(self)
+        utils.TestSetup.create_TestUser(self)
+        utils.check_or_try_logout_user(self)
+
+        path = s.SigninAPI.path
+        data_ok = {"user_name": self.test_user_name, "password": self.test_user_name}
+        data_bad = {"user_name": self.test_user_name, "password": "random-dont-care"}
+        for ctype in [CONTENT_TYPE_JSON, CONTENT_TYPE_HTML, CONTENT_TYPE_PLAIN]:
+            headers = {"Accept": ctype}
+            resp = utils.test_request(self, "GET", path, params=data_ok, headers=headers)
+            utils.check_response_basic_info(resp, 200, expected_method="GET", expected_type=ctype)
+
```

Comment:
Why is it not required to add a logout before testing another sign in here?

Reply:
Since login creates a cookie that overrides the previous one (same name/host), login again replaces it directly. 
I will add a logout for clarity of the intent, but it is not required.